### ISSUE TITLE
Gradle task is failing

### ISSUE THREAD

**Issue by vRallev** [KeepSafe/dexcount-gradle-plugin#217]
I'm using plugin `0.7.3`, the Android Gradle plugin `3.0.0-beta6`. We have multiple modules in our project and the dexcount task is failing for a library module when I run `./gradlew :libs:something:assembleDebug`
```
org.gradle.api.tasks.TaskValidationException: A problem was found with the configuration of task ':libs:something:countDebugDexMethods'.
        at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:55)
        at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:88)
        at org.gradle.api.internal.tasks.execution.ResolveTaskArtifactStateTaskExecuter.execute(ResolveTaskArtifactStateTaskExecuter.java:52)
        at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:52)
        at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:54)
        at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
        at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:34)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker$1.run(DefaultTaskGraphExecuter.java:248)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:336)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:328)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:197)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:107)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:241)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:230)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.processTask(DefaultTaskPlanExecutor.java:124)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.access$200(DefaultTaskPlanExecutor.java:80)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:105)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:99)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.execute(DefaultTaskExecutionPlan.java:625)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.executeWithTask(DefaultTaskExecutionPlan.java:580)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.run(DefaultTaskPlanExecutor.java:99)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
        at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:46)
        at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
Caused by: org.gradle.api.InvalidUserDataException: Directory '/Users/rwondratschek/dev/projects/evernote/evernote/libs/something/build/outputs/aar/something.aar' specified for property 'inputDirectory' is not a directory.
        at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:47)
        ... 23 more
```

Is this a known issue?

**Comment by benjamin-bader** [comment]
Not known by me!  This line is pretty suggestive:
> Caused by: org.gradle.api.InvalidUserDataException: Directory '/Users/rwondratschek/dev/projects/evernote/evernote/libs/something/build/outputs/aar/something.aar' specified for property 'inputDirectory' is not a directory.

Seems pretty reasonable that `something.aar` is _not_ a directory.  Is that something a typical (i.e. nothing fancy in the build.gradle file) library project?

**Comment by benjamin-bader** [comment]
Then perhaps something has changed in recent AS betas; in the alphas (and early betas, I think) the library task's output was a directory, not an .aar.  I'll take a look.  Thanks for the report!

**Comment by calvarez-ov** [comment]
I'm using `com.android.tools.build:gradle:2.3.3` (which I believe is the latest stable version) and I have an error using the dexcount plugin `0.8.0`.

```
* What went wrong:
A problem occurred configuring project ':libon-lite-client'.
> Failed to notify project evaluation listener.
   > com.android.build.gradle.api.ApkVariant.getOutputs()Lorg/gradle/api/DomainObjectCollection;
```

If I use dexcount version `0.7.3` it works fine.

Is the `0.8.0` incompatible with the android plugin `2.3.3`?

(Commenting here because I think my issue may be due to #218 which was done to fix this issue here)

**Comment by benjamin-bader** [comment]
@calvarez-ov two things:

1) Would you please re-run your build with `--stacktrace` and include the full trace?
2) Would you please open a new issue for this?

Thanks!

**Comment by benjamin-bader** [comment]
@calvarez-ov Actually, never mind - I reproduced the bug and fixed it.  Release 0.8.1 should work for you - if not, please open another issue.  Thanks for reporting this one!

**Comment by calvarez-ov** [comment]
@benjamin-bader You fixed it faster than I could reply :)  0.8.1 is working fine. Thanks!

**Comment by vRallev** [comment]
@benjamin-bader I'm sorry for the delay, but I was on vacation the last week. The new version is almost working fine. I found another bug and will file a separate issue.
